Content.Server/Chemistry/EntitySystems/SolutionTransferSystem.cs

```diff
@@ -0,0 +1,74 @@
+using Content.Shared.Verbs;
+using Content.Server.Chemistry.Components;
+using Content.Shared.Chemistry.EntitySystems;
+using JetBrains.Annotations;
+using Robust.Shared.GameObjects;
+using Content.Shared.Chemistry.Components.SolutionManager;
+using Robust.Shared.IoC;
+using Content.Shared.ActionBlocker;
+using Robust.Shared.Localization;
+using Content.Shared.Notification.Managers;
+using Robust.Server.GameObjects;
+using System.Collections.Generic;
+using Content.Shared.Chemistry.Reagent;
+
+namespace Content.Server.Chemistry.EntitySystems
+{
+	[UsedImplicitly]
+    public class SolutionTransferSystem : EntitySystem
+    {
+        [Dependency] private readonly ActionBlockerSystem _actionBlockerSystem = default!;
+
+        /// <summary>
+        ///     Default transfer amounts for the set-transfer verb.
+        /// </summary>
+        public static readonly List<int> DefaultTransferAmounts = new() { 1, 5, 10, 25, 50, 100, 250, 500, 1000};
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<SolutionTransferComponent, GetAlternativeVerbsEvent>(AddSetTransferVerbs);
+        }
+
+        private void AddSetTransferVerbs(EntityUid uid, SolutionTransferComponent component, GetAlternativeVerbsEvent args)
+        {
+            if (!args.CanAccess || !args.CanInteract || !component.CanChangeTransferAmount)
+                return;
+
+            if (!args.User.TryGetComponent<ActorComponent>(out var actor))
+                return;
+            
+            // Custom transfer verb
+            Verb custom = new("transfer:custom");
+            custom.Text = Loc.GetString("comp-solution-transfer-verb-custom-amount");
+            custom.Category = VerbCategory.SetTransferAmount;
```

Comment:
Currently categories are just instances of the VerbCategory class. There are just several static instances defined in the VerbCategory class itself. You can always create a new verb category instance when defining your verb, so they are already dynamic. But yeah, I don't know if there's a need for that.

The way they are done at the moment also means that instead of just sending a verb category key, each verb in the list has it's category instance serialized and sent over the network, which is wasteful if dynamic categories are never used.